src/util/BaseLoginController.js

```diff
@@ -104,6 +105,16 @@ define(['okta', 'q'], function (Okta, Q) {
       this.trigger('pageRendered', {page: this.className});
 
       this.trigger('afterRender', { controller: this.className });
+
+      this.listenTo(this, 'transactionError', function (err) {
+        if (!err.statusCode) {
+          // Bring the statusCode to the top-level of the Error
+          err.statusCode = err.xhr && err.xhr.status;
+        }
+        // Some controllers return the className as a function - process it here:
+        var className = typeof this.className === 'function' ? this.className() : this.className;
+        this.trigger('afterError', { error: err }, { controller: className });
```

Comment:
I am a little confused... we will trigger the event with 2 objects:
```
1: { error: { statusCode: code, message: 'some_message' } }
2: { controller: 'name' }
```
in the README we are saying that we will return 2 objects, an `error` and a `context`, so shouldn't we be returning
```
1: { statusCode: code, message: 'some_message' },
2: { controller: 'name' }
```
It seems to me we don't need to put the error object inside an `error` property, but just return it itself...
```
this.trigger('afterError', err, { controller: className });
```

Reply:
You're correct! This was a *rookie* mistake on my end. Fixed with the next commit.